test/plugin_reload_test.go

```diff
@@ -0,0 +1,103 @@
+package test
+
+import (
+	"bytes"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/mholt/caddy"
+)
+
+func TestPluginReload(t *testing.T) {
+	tests := []struct {
+		corefile    string
+		corefileNew string
+		expected    string
+	}{
+		{`.:0 {
+			reload 2s
+			whoami
+			health localhost:0
+		}
+			`,
+			`.:0 {
+			reload 2s
+			whoami
+			health localhost
+		}
+			`,
+			"Corefile successfully loaded",
```

Comment:
As you do a strings.Contains I would makes this as short as possible, probably just `successfully` to prevent simple log changes making this test fail.